test/tasks_test.rb

```diff
@@ -79,4 +79,24 @@ def capture_stderr
       out.must_include "test/helper.rb"
     end
   end
+
+  describe "abort_if_pending_migrations" do
+    before do
+      rake('db:create')
+    end
+
+    it "passes when there is no pending migrations" do
+      ActiveRecord::Migrator.any_instance.stubs(:pending_migrations).returns([])
+      out = capture_stderr { rake('db:abort_if_pending_migrations') }
+      out.must_be_empty
+    end
+
+    it "fails when migrations are pending" do
+      ActiveRecord::Migrator.any_instance.stubs(:pending_migrations).returns([stub(version: 1, name: 'Fake')])
+      out = capture_stderr do
+        rake('db:abort_if_pending_migrations') rescue nil # Catch SystemExit
```

Comment:
`SystemExit` is a direct descentant of `Exception` and won’t be caught by `rescue`. So perhaps it can just be removed?

Reply:
It cannot just be removed. `abort_if_pending_migrations` -> `abort` -> `exit` -> `raise SystemExit`.